src/assets/js/react/components/FontManager/AddFont.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import FontVariant from './FontVariant'
+import Kashida from './Kashida'
+import AddFontFooter from './AddFontFooter'
+
+export const AddFont = (
+  {
+    style,
+    id,
+    fontList,
+    label,
+    fontId,
+    onHandleInputChange,
+    onHandleKashidaChange,
+    onHandleUpload,
+    onHandleDeleteFontStyle,
+    onHandleCancelEditFont,
+    onHandleSubmit,
+    fontStyles,
+    kashida,
+    validateLabel,
+    validateRegular,
+    msg,
+    loading,
+    tabIndexFontName,
+    tabIndexFontFiles,
+    tabIndexFooterButtons
+  }
+) => (
+  <div className={style}>
+    <form onSubmit={onHandleSubmit}>
+      <h2>{id ? 'Update Font' : 'Add Font'}</h2>
+      <p>
+        {id ? 'Once saved, PDFs configured to use this font will have your changes applied automatically for newly-generated documents.' : 'Install new fonts for use in your PDF documents.'}
+      </p>
+
+      <label htmlFor='gfpdf-add-font-name-input'>
+        Font Name <span className='required'>(required)</span>
+      </label>
+      <p>The font name can only contain alphanumeric characters or spaces.</p>
+      <input
+        type='text'
+        id='gfpdf-add-font-name-input'
+        className={!validateLabel ? 'input-label-validation-error' : ''}
+        name='label'
+        value={label}
+        maxLength='60'
+        onChange={onHandleInputChange}
+        tabIndex={tabIndexFontName}
+      />
+      {!validateLabel && <span className='required'><em>validation error</em></span>}
+
+      <label>Font Files</label>
+      <p>
+        Select or drag and drop your .ttf font file for the variants below. Only the Regular type is required.
+      </p>
+
+      <FontVariant
+        fontStyles={fontStyles}
+        validateRegular={validateRegular}
+        onHandleUpload={onHandleUpload}
+        onHandleDeleteFontStyle={onHandleDeleteFontStyle}
+        msg={msg}
+        tabIndexFontFiles={tabIndexFontFiles}
+      />
+
+      {(id && fontList) && fontList.map(font => {
```

Comment:
@debbieellis7 wouldn't it be better to pass through the contents of `font` to this component, instead of the individual values `label`, `kashida` etc? You don't have to loop through the entire fontList then.